### ISSUE TITLE
missing variable re-initialization during inlining

### ISSUE THREAD

**Issue by vtjnash** [JuliaLang/julia#12620]
inlining can get the the NewvarNode wrong for some functions, resulting in missed errors or segfaults:
```julia
@inline function foo(x)
         if x
           y = 2
         else
           return y
         end
       end
function bar()
         for i = 1:3 
           foo(i==1)
         end
       end

julia> bar()

julia> 
```
expected answer:
```julia
julia> bar()
ERROR: UndefVarError: y#7057 not defined
 in bar at ???
```

**Comment by damiendr** [comment]
Just happened onto this bug.

The problem seems to be that `NewvarNode(:y)` is not escaped like the other symbols:
```julia
@inline function testnew(count)
    x = 1
    i = 0
    while i <= count
        y = x + y
        x = y
        i += 1
    end
end
function test(a)
    testnew(a)
end
println(@code_typed test(10))
test(10)
```
```
Any[:($(Expr(:lambda, Any[:a], Any[Any[Any[:a,Int64,0],Any[symbol("##x#7179"),Int64,2],Any[symbol("##i#7180"),Int64,2],Any[symbol("##y#7181"),Union{},50]],Any[],Any[],Any[]], :(begin  # /Users/x/test_inline.jl, line 11: # /Users/x/test_inline.jl, line 2:
        ##x#7179 = 1 # /Users/x/test_inline.jl, line 3:
        ##i#7180 = 0 # /Users/x/test_inline.jl, line 4:
        NewvarNode(:y)
        unless (Base.sle_int)(##i#7180::Int64,a::Int64)::Bool goto 3
        1:  # /Users/x/test_inline.jl, line 5:
        ##y#7181 = ##x#7179::Int64 + ##y#7181::Union{}::Union{} # /Users/x/test_inline.jl, line 6:
        ##x#7179 = ##y#7181::Union{} # /Users/x/test_inline.jl, line 7:
        ##i#7180 = (Base.box)(Base.Int,(Base.add_int)(##i#7180::Int64,1))
        2:
        unless (Base.box)(Base.Bool,(Base.not_int)((Base.sle_int)(##i#7180::Int64,a::Int64)::Bool)) goto 1
        3:
        4:
        return
    end::Void))))]
ERROR: LoadError: UndefVarError: y#7181 not defined
 [inlined code] from /Users/x/test_inline.jl:5
 in test at /Users/x/test_inline.jl:11
 in include at ./boot.jl:261
 in include_from_node1 at ./loading.jl:392
 in process_options at ./client.jl:277
 in _start at ./client.jl:377
while loading /Users/x/test_inline.jl, in expression starting on line 14
```

**Comment by damiendr** [comment]
OK, I have a fix almost ready. Making a test for it now.

**Comment by JeffBezanson** [comment]
fixed by #14014
